code/model.py

```diff
@@ -1058,23 +1117,17 @@ def forward(self, batch):
             if polarity_layers > 0:
                 polarity_output = self.components["polarity"]["cnn_sequential"](polarity_output)
 
-            # predicted prob of B or I in target and expression outputs
-            bi_probs = (target_output[:,:,1:].sum(dim=-1) + expression_output[:,:,1:].sum(dim=-1))/2.
-            bi_probs = bi_probs[:,:polarity_output.size(2)]
-            values =  torch.mul(
-                polarity_output.permute(1, 0, 2), # embedding, batch, sequence
-                bi_probs,
-            ).permute(2, 1, 0)  # sequence, batch, embedding
-
-            polarity_output = polarity_output.permute(2, 0, 1)  # sequence, batch, embedding
-            polarity_output, _ = self.components["polarity"]["attention"](
-                polarity_output,  # query, i.e. polar cnn output w/ weights
-                polarity_output,  # keys, i.e. (polar cnn output).T for self attention
-                values,  # values include probabilities for B and I tags
-                need_weights=False,
-                # TODO: implement attention mask?
-            )
-            polarity_output = polarity_output.permute(1, 2, 0)  # batch, embedding, sequence
+            # attention block
+            # values = polarity_output.permute(2, 0, 1)
+            # polarity_output = polarity_output.permute(2, 0, 1)  # sequence, batch, embedding
+            # polarity_output, _ = self.components["polarity"]["attention"](
+            #     polarity_output,  # query, i.e. polar cnn output w/ weights
+            #     polarity_output,  # keys, i.e. (polar cnn output).T for self attention
+            #     values,  # values should include probabilities for B and I tags
+            #     need_weights=False,
+            #     # TODO: implement attention mask?
+            # )
+            # polarity_output = polarity_output.permute(1, 2, 0)  # batch, embedding, sequence
```

Comment:
Currently removed.

`bi_probs` wasn't exactly what the `gold_probs` from the IMN project. 
They used a guided warm-up, where every epoch plays  `p(epochs) = 5/exp(epochs+5)` of a role in these probabilities. As epochs increase, the affect these gold labels that's allow into the attention mech. decreases "exponentially".  

`bi_probs` only took the logits of the `B` and `I` predictions form previous subtask.